site/src/components/Resources/AppLink/AppLink.tsx

```diff
@@ -93,75 +93,65 @@ export const AppLink: FC<AppLinkProps> = ({ app, workspace, agent }) => {
 
   const isPrivateApp = app.sharing_level === "owner";
 
-  const button = (
-    <AgentButton
-      startIcon={icon}
-      endIcon={isPrivateApp ? undefined : <ShareIcon app={app} />}
-      disabled={!canClick}
-    >
-      <span
-        css={
-          !isPrivateApp && {
-            marginRight: 8,
-          }
-        }
-      >
-        {appDisplayName}
-      </span>
-    </AgentButton>
-  );
-
   return (
     <Tooltip title={primaryTooltip}>
-      <span>
-        <Link
-          href={href}
-          target="_blank"
-          css={{
-            pointerEvents: canClick ? undefined : "none",
-            textDecoration: "none !important",
-          }}
-          onClick={
-            canClick
-              ? async (event) => {
-                  event.preventDefault();
-                  // This is an external URI like "vscode://", so
-                  // it needs to be opened with the browser protocol handler.
-                  if (app.external && !app.url.startsWith("http")) {
-                    // If the protocol is external the browser does not
-                    // redirect the user from the page.
+      <Link
+        color="inherit"
+        component={AgentButton}
+        startIcon={icon}
+        endIcon={isPrivateApp ? undefined : <ShareIcon app={app} />}
+        disabled={!canClick}
+        href={href}
+        target="_blank"
+        css={{
+          pointerEvents: canClick ? undefined : "none",
+          textDecoration: "none !important",
+        }}
+        onClick={
+          canClick
```

Comment:
I'm wondering if, just to simplify the nesting a little bit, this could be turned into this:
```tsx
onClick={async (event) => {
  if (!canClick) {
    return;
  }

  // Rest of the logic
}}
```

Reply:
I do agree!